src/api/channels.coffee

```diff
@@ -198,42 +160,33 @@ processPostDeleteTriggers = (channel) ->
 ###
 # Deletes a specific channels details
 ###
-exports.removeChannel = `function *removeChannel(channelId) {
-
-  // Test if the user is authorised
-  if (authorisation.inGroup('admin', this.authenticated) === false) {
-    logger.info('User ' +this.authenticated.email+ ' is not an admin, API access to removeChannel denied.')
-    this.body = 'User ' +this.authenticated.email+ ' is not an admin, API access to removeChannel denied.'
-    this.status = 'forbidden';
-    return;
-  }
-
-  // Get the values to use
-  var id = unescape(channelId);
-
-  try {
-    var numExistingTransactions = yield Transaction.count({ channelID: id }).exec();
-
-    // Try to get the channel (Call the function that emits a promise and Koa will wait for the function to complete)
-    var channel;
-    if (numExistingTransactions === 0) {
-      //safe to remove
-      channel = yield Channel.findByIdAndRemove(id).exec();
-    } else {
-      //not safe to remove. just flag as deleted
-      channel = yield Channel.findByIdAndUpdate(id, { status: 'deleted' }).exec();
-    }
-
-    // All ok! So set the result
-    this.body = 'The channel was successfully deleted';
-    logger.info('User %s removed channel with id %s', this.authenticated.email, id);
-
-    processPostDeleteTriggers(channel);
-  }
-  catch (e) {
-    // Error! So inform the user
-    logger.error('Could not remove channel by id: ' +id+ ' via the API: ' + e);
-    this.body = e.message;
-    this.status = 'internal server error';
-  }
-}`
+exports.removeChannel = (channelId) ->
+
+  # Test if the user is authorised
+  if authorisation.inGroup('admin', this.authenticated) is false
+    utils.logAndSetResponse this, 'forbidden', "User #{this.authenticated.email} is not an admin, API access to removeChannel denied.", 'info'
+    return
+
+  # Get the values to use
+  id = unescape channelId
+
+  try
+    numExistingTransactions = yield Transaction.count({ channelID: id }).exec()
+
+    # Try to get the channel (Call the function that emits a promise and Koa will wait for the function to complete)
+    channel
```

Comment:
This can be removed. It'll just translate to a one line `channel;` in js, not a var declaration.